src/reconcile/rec_write.c

```diff
@@ -1436,14 +1436,21 @@ __wt_rec_split_finish(WT_SESSION_IMPL *session, WT_RECONCILE *r)
  *     Move a saved WT_UPDATE list from the per-page cache to a specific block's list.
  */
 static int
-__rec_supd_move(WT_SESSION_IMPL *session, WT_MULTI *multi, WT_SAVE_UPD *supd, uint32_t n)
+__rec_supd_move(
+  WT_SESSION_IMPL *session, WT_MULTI *multi, WT_SAVE_UPD *supd, uint32_t n)
 {
     uint32_t i;
 
+    multi->supd_restore = false;
+
     WT_RET(__wt_calloc_def(session, n, &multi->supd));
 
-    for (i = 0; i < n; ++i)
+    for (i = 0; i < n; ++i) {
+        if (supd->restore)
```

Comment:
You can just apply it to the multi here. No need to do the if check.

Reply:
Can I? multi->supd_restore is an aggregation of supd->restore. If the first supd->restore is true and the second is false, we will overwrite the multi->supd_restore to the wrong value.

Reply:
multi->supd_restore should be true if anyone of the saved updates needs to be restored.

Reply:
Oh good point, my bad.